### PR TITLE
docs: Add example for health_upstream

### PR DESCRIPTION
**Pull request by dylanschultzie**
Relevant PR: https://github.com/caddyserver/caddy/pull/6451

Very much open to feedback on verbiage and the example provided.

### CODE REVIEW

**Review by mholt — APPROVED** [review]
Makes sense to me. :+1: Thanks for the docs!

**Comment by dylanschultzie** [comment]
@mholt are you able to merge?

**Comment by francislavoie** [comment]
We don't merge docs changes until the change is released in Caddy. We will after release.

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md [hunk 1]
```diff
@@ -852,3 +855,16 @@ example.com {
...
+Using [active health checks](#active-health-checks) and `health_upstream` can be helpful when creating an intermediate service to do a more thorough health check. `{http.reverse_proxy.active.target_upstream}` can then be used as a header to provide the original upstream to the health check service.
+
+```caddy
+example.com {
+	reverse_proxy node1:80 node2:80 node3:80 {
+		health_uri /health
+        health_upstream 127.0.0.1:53336
+        health_headers {
+          Full-Upstream {http.reverse_proxy.active.target_upstream}
+        }
```
Comment (mohammed90) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md]: The indentation here isn't aligned.

**Review by mohammed90 — CHANGES_REQUESTED** [review]

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md (hunk 1 above)
Comment (dylanschultzie) [reply src/docs/markdown/caddyfile/directives/reverse_proxy.md]: Fixed, thanks!

Comment (francislavoie) [reply src/docs/markdown/caddyfile/directives/reverse_proxy.md]: The indentation is still wrong. Use tabs, not spaces.

Comment (dylanschultzie) [reply src/docs/markdown/caddyfile/directives/reverse_proxy.md]: Hmm. I did use tabs, the browser editor must be auto correcting it.

**Comment by mholt** [comment]
(Apologies; I'm the main reason it's been slow to release this summer!)

**Comment by francislavoie** [comment]
Merging into the 2.9 branch. Thanks!

**Pull request merged**